src/main/java/com/streamr/broker/stats/EventsStats.java

```diff
@@ -0,0 +1,162 @@
+package com.streamr.broker.stats;
+
+import com.streamr.client.protocol.message_layer.StreamMessage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.concurrent.atomic.AtomicLong;
+
+public abstract class EventsStats implements Stats {
+    protected static final Logger log = LogManager.getLogger();
+    private static final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
SimpleDateFormat is not synchronized!

"
* <p>
 * Date formats are not synchronized.
 * It is recommended to create separate format instances for each thread.
 * If multiple threads access a format concurrently, it must be synchronized
 * externally.
"